packages/strapi/lib/core/load-plugins.js

```diff
@@ -37,8 +37,11 @@ const loadLocalPlugins = async ({ dir }) => {
     loadFiles(pluginsDir, '{*/!(config)/*.*(js|json),*/package.json}'),
     loadConfig(pluginsDir, '*/config/**/*.+(js|json)'),
   ]);
-
-  return _.merge(files, configs);
+  const userConfigs = Object.keys(files).reduce((acc, plugin) => {
+    acc[plugin] = config.get(['plugins', plugin], {});
```

Comment:
this is not gonig to be merged in the right place  like that.

You need to put the config in a config property here:
```js
acc[plugin] = { config: config.get() }
```

Reply:
Done.